client/scripts/views/pages/community_cards.tsx

```diff
@@ -161,34 +155,109 @@ class HomepageCommunityCards implements m.ClassComponent {
           if (Array.isArray(entity)) {
             const [chain, nodeList]: [string, any] = entity as any;
             return m(ChainCard, { chain, nodeList });
-          } else if (entity.id) {
-            return m(CommunityCard, { community: entity });
           }
           return null;
         });
+      return res;
+    };
 
-    const sortedChainsAndCommunities = sortChainsAndCommunities(
-      myChains.filter((c) => c[1][0] && !c[1][0].chain.collapsedOnHomepage)
+    const sortedChains = sortChains(
+      myChains.filter((c) => c[1][0] && !c[1][0].chain.collapsedOnHomepage),
+      this.filters
     );
-    const betaChainsAndCommunities = sortChainsAndCommunities(
-      myChains.filter((c) => c[1][0] && c[1][0].chain.collapsedOnHomepage)
+    const betaChains = sortChains(
+      myChains.filter((c) => c[1][0] && c[1][0].chain.collapsedOnHomepage),
+      this.filters
     );
 
     const totalCommunitiesString = buildCommunityString(
-      sortedChainsAndCommunities.length + betaChainsAndCommunities.length
+      sortedChains.length + betaChains.length
     );
 
     return (
       <div class="HomepageCommunityCards">
-        <div class="communities-header">{totalCommunitiesString}</div>
-        <div class="communities-list">{sortedChainsAndCommunities}</div>
+        <div class="header-section">
+          <div class="communities-header">{totalCommunitiesString}</div>
+          <div class="filter-buttons">
+            <CWButton
```

Comment:
all of these buttons should be algorithmically generated from the 3 arrays of strings. `for cat in categories, return button(cat); for base in bases...; etc`